tests/test_utils.py

```diff
@@ -4,17 +4,39 @@
 from pathlib import Path
 from unittest.mock import patch
 
+import isceobj  # noqa
 import numpy as np
+import pytest
 from osgeo import gdal
 
 import hyp3_isce2.utils as utils
 
-import isceobj  # noqa
-
 
 gdal.UseExceptions()
 
 
+def test_check_older_granule_is_reference():
+    utils.check_older_granule_is_reference(
+        'S1_000000_IW1_20200101T000000_VV_0000-BURST', 'S1_000000_IW1_20200201T000000_VV_0000-BURST'
+    )
+
+    utils.check_older_granule_is_reference(
+        reference=['S1_000000_IW1_20200101T000000_VV_0000-BURST', 'S1_000001_IW1_20200101T000000_VV_0000-BURST'],
+        secondary=['S1_000000_IW1_20200201T000000_VV_0000-BURST', 'S1_000001_IW1_20200201T000000_VV_0000-BURST'],
+    )
+
+    with pytest.raises(ValueError, match='Reference granules must be from one date*'):
```

Comment:
Since the secondary granules are the ones that actually differ in this test case, it's slightly confusing for the error message pattern to mention reference granules.
```suggestion
    with pytest.raises(ValueError, match='.* granules must be from one date .*'):
```